include/static-pages.php

```diff
@@ -105,6 +105,21 @@ public function init() {
 	 */
 	protected function get_translation( $static_page, $language ) {
 		$translations = $this->model->post->get_raw_translations( $this->$static_page );
+
+		if ( empty( $translations ) ) {
+			$page_lang = $this->model->post->get_object_term( $this->$static_page, $this->model->post->get_tax_language() );
+
+			if ( empty( $page_lang ) || $page_lang->slug !== $language['slug'] ) {
+				return 0;
+			}
+
+			/*
+			 * Current page doesn't have any translation,
+			 * but has a language defined corresponding to the current one.
+			 */
+			return $this->$static_page;
+		}
```

Comment:
I propose something slightly simpler with a different comment:
```suggestion
		// When the current static page doesn't have any translation, we must return itself for its language.
		if ( empty( $translations ) ) {
			$page_lang = $this->model->post->get_object_term( $this->$static_page, $this->model->post->get_tax_language() );

			if ( ! empty( $page_lang ) && $page_lang->slug === $language['slug'] ) {
				return $this->$static_page;
			}
		}
```